src/map.rs

```diff
@@ -888,8 +995,44 @@ where
     ///
     /// ```
     /// use hashbrown::HashMap;
+    ///
     /// let mut map: HashMap<&str, isize> = HashMap::new();
+    /// // Map is empty and doesn't allocate memory
+    /// assert_eq!(map.capacity(), 0);
+    ///
     /// map.try_reserve(10).expect("why is the test harness OOMing on 10 bytes?");
+    ///
+    /// // And now map can hold at least 10 elements
+    /// assert!(map.capacity() >= 10);
+    /// ```
+    /// If the capacity overflows, or the allocator reports a failure, then an error
+    /// is returned:
+    /// ```
+    /// # fn test() {
+    /// use hashbrown::HashMap;
+    /// use hashbrown::TryReserveError;
+    /// let mut map: HashMap<i32, i32> = HashMap::new();
+    ///
+    /// match map.try_reserve(usize::MAX) {
+    ///     Err(error) => match error {
+    ///         TryReserveError::CapacityOverflow => {}
+    ///         _ => panic!("TryReserveError::AllocError ?"),
+    ///     },
+    ///     _ => panic!(),
+    /// }
+    ///
+    /// match map.try_reserve(usize::MAX / 64) {
+    ///     Err(error) => match error {
+    ///         TryReserveError::AllocError { .. } => {}
+    ///         _ => panic!("TryReserveError::CapacityOverflow ?"),
+    ///     },
+    ///     _ => panic!(),
+    /// }
```

Comment:
I would just remove this, it's very difficult to get it to trigger reliably and can cause OOM errors in other processes on some systems.

Reply:
But if I delete, then there are will not be shown all variants of the error.  This is just an example, no one will actually try to allocate 16 million terabytes of memory in a real program :-)

Reply:
Can I just write a comment that running this part of the example could cause OOM errors in other processes on some systems?

Reply:
The examples are tested by CI, so this would cause CI tests to fail. Maybe you can mark the example as ````no_run` so that it isn't executed when running tests.